File: src/common/web/features/ProjectFeaturesCatalog.ts
import { ItemsCatalog } from "../utils/ItemsCatalog";
import { ProjectFeature, ProjectFeatureFlags } from "./ProjectFeature";

/**
 * Global catalog of all registered backend types.
 *
 * This is a globally accessible list of all project features, associated with their respective IDs.
 */
@ItemsCatalog.define()
export class ProjectFeaturesCatalog extends ItemsCatalog<ProjectFeature> {
    public static filter(flags: ProjectFeatureFlags): ProjectFeature[] {
        return Object.values(this.items).filter((feature) => feature.hasFlags(flags));
    }
}

File: src/common/py/features/project_features_catalog.py
from .project_feature import ProjectFeature
from ..utils import ItemsCatalog


@ItemsCatalog.define()
class ProjectFeaturesCatalog(ItemsCatalog[ProjectFeature]):
    """
    Global catalog of all registered backend types.

    This is a globally accessible list of all project features, associated with their respective IDs.
    """
